### PR TITLE
ibm: p10bmc: Add I2C mux workarounds

### PR DESCRIPTION
**Pull request by eddiejames**
On Everest systems, some BMC cards need various I2C muxes reset
before they will work. Add a script to reset the muxes and probe
the necessary drivers afterwards.

Signed-off-by: Eddie James <eajames@linux.ibm.com>

### CODE REVIEW

**Comment by geissonator** [comment]
add to approvelist

**Comment by geissonator** [comment]
jenkins run tests please

**Comment by geissonator** [comment]
jenkins run tests please

File: meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service
Comment (geissonator) [inline meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service:13]: basic.target is not something we use often, I guess you're doing this because you want this to run before any of hte other normal user space apps from the multi-user.target?

Comment (eddiejames) [reply meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service:13]: That's correct, especially before any users start grabbing GPIOs

File: meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service
Comment (eddiejames) [reply meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service:8]: Ah ok, will change

File: meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh
Comment (eddiejames) [reply meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh:4]: I'm not sure, I guess it would be in the VPD but I think this should run before VPD collection gets going... guess I could test afterwards, but still not sure how to check the right VPD?

Comment (geissonator) [reply meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh:4]: I take it there's no adverse effects if we run it against hardware that doesn't need it? Waiting for vpd collection is probably going to add more complexity then we want.

File: meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service
Comment (amboar) [reply meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service:13]: Can we not just express the actually dependencies rather than hand-waving with `basic.target`?

File: meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh
Comment (amboar) [inline meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh:13]: Can we please put these `i2cset` calls in a named function? Otherwise the intent isn't clear and the code's a bit impenetrable. Applies to the remaining `i2cset` calls below.

File: meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh
Comment (amboar) [inline meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh:2]: Can we please add `set -eu`?

Comment (eddiejames) [reply meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh:2]: Sure

File: meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service
Comment (eddiejames) [reply meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service:13]: There are no dependencies though

File: meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh
Comment (eddiejames) [reply meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.sh:13]: Sure

File: meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service
Comment (geissonator) [inline meta-ibm/recipes-phosphor/chassis/mux-workarounds/mux-workarounds.service:8]: nit: This isn't needed anymore since you now have the full path to the script. Not a big deal though.

**Comment by geissonator** [comment]
Another nit is that since we're doing the full review in github, lets follow github review process and create a new commit for code review comments. It makes it easy to see the diff between patches. We can just squash down to one commit when we merge.

**Review by geissonator — APPROVED** [review]
Lets wait for a +1 from @amboar before merging.

**Pull request closed without merging**
